lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs

```diff
@@ -90,8 +102,59 @@ spec = do
         it "throws when length (key) != 32" $
             evaluate (singleAddressFromKey (Proxy @'Mainnet) (XPub "\148" cc))
             `shouldThrow` anyErrorCall
+    describe "keyToAddress is consistent with jcli" $ do
+        -- The goldens have been verified with jcli as follows:
+        -- $ jcli key generate --type=Ed25519 | jcli key to-public
+        -- ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c
+        -- $ jcli address single ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c --testing
+        -- ta1sv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4wwywkd6j
+        -- $ jcli address single ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c
+        -- ca1qv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4ww0xfchg
+
+        bech32KeyToAddrGolden
+            "ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c"
+            "ta1sv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4wwywkd6j"
+            "ca1qv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4ww0xfchg"
```

Comment:
A caveat is that it is unreadable if you don't know of the meaning of the human readable parts.

A benefit from passing in `Text` directly is that `bech32KeyToAddrGolden` has more control, and can output the key in bech32 format despite there being no encodeBech32Key.

Maybe we should add `encode/decodeBech32Key` in `AddressDerivation`… 🤔

Reply:
To make things more readable without impeding too much on the flexibility, you may use a record type and use record fields at the call-site :+1: